src/eve/datamodels.py

```diff
@@ -0,0 +1,1008 @@
+                    RuntimeWarning,
+                )
+            else:
+                reference_module_globals[class_name] = concrete_cls
+
+    return concrete_cls
+
+
+def validator(name: str) -> Callable[[Callable], Callable]:
+    """Decorator to define a custom field validator for a specific field.
+
+    Arguments:
+        name: Name of the field to be validated by the decorated function.
+    """
+    assert isinstance(name, str)
+
+    def _field_validator_maker(func: Callable) -> Callable:
+        setattr(func, _FIELD_VALIDATOR_TAG, name)
+        return func
+
+    return _field_validator_maker
+
+
+def root_validator(func: Callable, /) -> classmethod:
+    """Decorator to define a custom root validator."""
+
+    cls_method = classmethod(func)
+    setattr(cls_method, _ROOT_VALIDATOR_TAG, None)
+    return cls_method
+
+
+def fields(
+    datamodel: Union[DataModelLike, Type[DataModelLike]], *, as_dataclass: bool = False
+) -> utils.FrozenNamespace:
+    """Return the field meta-information of a Data Model.
+
+    Arguments:
+        datamodel: A Data Model class or instance.
+
+    Keyword Arguments:
+        as_dataclass: If ``True`` (the default is ``False``), field information is returned
+            as :class:`dataclass.Field` instances instead of :class:`attr.Attribute`.
+
+    Examples:
+        >>> from typing import List
+        >>> @datamodel
+        ... class Model:
+        ...     amount: int = 1
+        ...     name: str
+        ...     numbers: List[float]
+        >>> fields(Model)  # doctest:+ELLIPSIS
+        FrozenNamespace(amount=Attribute(name='amount', default=1, ...),\
+ name=Attribute(name='name', default=NOTHING, ...),\
+ numbers=Attribute(name='numbers', default=NOTHING, ...))
+        >>> fields(Model, as_dataclass=True)  # doctest:+ELLIPSIS
+        FrozenNamespace(amount=Field(name='amount',type='int',default=1,...),\
+ name=Field(name='name',type='str',default=<dataclasses._MISSING_TYPE object ...),\
+ numbers=Field(name='numbers',type='List[float]',default=<dataclasses._MISSING_TYPE object ...))
+
+    """
+    if not is_datamodel(datamodel):
+        raise ValueError(f"Invalid datamodel instance or class: '{datamodel}'.")
+
+    if as_dataclass:
+        return utils.FrozenNamespace(**{f.name: f for f in datamodel.__dataclass_fields__})
+    else:
+        ns = getattr(datamodel, _MODEL_FIELDS)
+        assert isinstance(ns, utils.FrozenNamespace)
+        return ns
+
+
+def field(
+    *,
+    default: Any = NOTHING,
+    default_factory: Callable[[None], Any] = NOTHING,
+    converter: Callable = None,
+    init: bool = True,
+    repr: bool = True,  # noqa: A002   # shadowing 'repr' python builtin
+    hash: Optional[bool] = None,  # noqa: A002   # shadowing 'hash' python builtin
+    compare: bool = True,
+    metadata: Optional[Mapping[Any, Any]] = None,
+) -> Any:  # attr.s lies on purpose in some typings
+    """Return an object to identify dataclass fields.
```

Comment:
Yes, but copying the interface (and the docstrings) from the equivalent `dataclasses.field`, since the idea is to make `datamodels` to look like `dataclasses` as much as possible and adding strict type checking. I will update the docstrings to make it more clear and add an example.

Reply:
Changed docstring and added an example.